lib/core/nuxt.js

```diff
@@ -105,11 +105,30 @@ export default class Nuxt {
     }
   }
 
-  addObjectHooks(hooksObj) {
-    Object.keys(hooksObj).forEach((name) => {
-      let hooks = hooksObj[name]
-      hooks = Array.isArray(hooks) ? hooks : [hooks]
-      hooks.forEach(hook => this.hook(name, hook))
+  getObjectHooks(hooksObj, keyList = [], stack = '') {
+    Object.keys(hooksObj).forEach((key) => {
+      if (typeof hooksObj[key] === 'object' && hooksObj[key] !== null) {
+        this.getObjectHooks(hooksObj[key], keyList, `${stack}:${key}`)
+      } else {
+        const value = hooksObj[key]
+        key = `${stack}:${key}`
+        keyList.push({
+          name: key.slice(1),
+          hook: value
+        })
+      }
+    })
+    return keyList.reduce((hash, pair) => {
```

Comment:
Convert an array to an object here, and then loop the object as an array in `addObjectHooks`, does it intend to ? How about use `keysList` as an object or just return array ?

Reply:
>  How about use keysList as an object or just return array ?

I would probably have to rework the recursion -- not sure. I think it's mostly fine as-is.